apps/src/code-studio/components/progress/LessonProgress.jsx

```diff
@@ -71,6 +72,10 @@ class LessonProgress extends Component {
       }
     }
 
+    if (this.props.isLabLoading !== nextProps.isLabLoading) {
```

Comment:
can you add a comment as to why this check is needed?

Reply:
Does the code make sense in the context of the `shouldComponentUpdate` function?  Basically everything here is just checking changed props that should result in doing a re-render...

Reply:
I guess I'm wondering if there's ever a case where we would still want to render a change, even if the lab loading state hasn't changed. Maybe not? But worth explaining why.

Reply:
fwiw, we will still render a change for other props changing.  This just makes sure we also render a change if the loading state changes.

Reply:
ah gotcha, makes sense!